src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -105,37 +96,49 @@ class JooqCalibrateDataRepository: CalibrateDataRepository
         return arrayList
     }
 
-    @Suppress("MagicNumber")
+    @Suppress("UnsafeCallOnNullableType")
     private fun getFilteredDataFromConnection(
         conn: Connection,
-        filterQuery: FilterQuery): List<CalibrateResultRow> {
-        val resultSet: ResultSet
-        var query = FILTER_TEMPLATE
-
-        for ((field, values) in filterQuery) {
-            if (values.size > 0) {
-                val questionMarks = values.map { "?" }.joinToString(",")
-                query += "$field IN ($questionMarks) AND "
-            }
-        }
-        query = query.dropLast(5)
-
-        val stmt: PreparedStatement = conn.prepareStatement(query)
-        var stmtIndex = 1;
-        for ((_, values) in filterQuery) {
-            values.forEach { it ->
-                if (it is String) {
-                    stmt.setString(stmtIndex, it)
-                } else if (it is Int) {
-                    stmt.setInt(stmtIndex, it)
+        filterQuery: FilterQuery): List<CalibrateResultRow>
+    {
+        val dataSource = SingleConnectionDataSource(conn)
+        return Database.connect(dataSource)
+            .from(ResultData)
+            .select()
+            .whereWithConditions {
+                if (filterQuery.indicator.size > 0) {
+                    it += ResultData.indicator inList filterQuery.indicator
+                }
```

Comment:
Could you make this a little less verbose with a util, something like:
```
fun conditionalAddWhere(conditions: MutableList, queryConfition: List<String>, column: Column)
{
    if (queryCondition.size > 0) {
        conditions += column inList queryConfition
    }
}
```
then call with 
```
.whereWithConditions {
    conditionalAddWhere(it, filterQuery.indicator, ResultData.indicator)
   ...etc
```